src/core/linter-rules/check-internal-slots.js

```diff
@@ -41,9 +43,12 @@ function linterFunction(conf, doc) {
 
 export const rule = new LinterRule(name, linterFunction);
 
-function checkElement({ previousSibling, textContent }) {
-  const potentialSlot = previousSibling.textContent + textContent;
-  // starts with dot, followed by [[ and any word
-  return !/^\.\[\[\w+\]\]/.test(potentialSlot);
-  // return true if regex test fails
+function filterElement({ textContent }) {
```

Comment:
This is great, the only thing I suggest is that we should check if ```previousSibling``` is not of type VAR, is the ```textContent``` of  ```previousSibling``` equal to "." or not.

```
const offendingElements = [...document.querySelectorAll("var+a")].filter(
  ({ previousSibling: { nodeName, textContent: previousText }, textContent }) => {
    const isInternalSlot = /\[\[\w+\]\]/.test(textContent.trim());
    const isPrevVar = nodeName && nodeName === "VAR" ? true : previousText  !==  ".";
    return isInternalSlot && isPrevVar;
  }
);
```

Reply:
I tried this on the following cases
```
<var>foo</var>bar<a>Foo</a>
<var>foo</var><a>[[bar]]</a>
<var>foo</var>Foo<a>[[bar]]</a>
<var>foo</var>.<a>[[bar]]</a>
```
Worked fine and the second and third elements were offending

Reply:
> This is great, the only thing I suggest is that we should check if previousSibling is not of type VAR, is the textContent of previousSibling equal to "." or not.

That might be problematic, because of this kind of sentence that is totally valid:

```JS
<p>... and the <var>foo</var>. However, the internal slot <a>[[bar]]</a> bla bla....</p>
```

(See https://www.w3.org/TR/payment-request/ for uses of internal slots, search for "[[" and you will see quite a few - though the situation above doesn't appear, it's definitely possible that it could appear in other specs.)

So, the only violation we can really check of is if `<a>[[internalSlot]]`</a> is immediately preceded by `<var>` in which case, there will be no `TextNode`:

```JS
<!-- no TextNode in between these two -->  
<var>foo</var><a>[[bar]]</a>
```

Reply:
I am a little confused, if ``` <var>foo</var>. However, the internal slot <a>[[bar]]</a>``` is a valid case, then the rule we created before is wrong because the only check was to match a "." before the "[[". Sp, the only invalid case is if there is no element between the ```<var>``` and ```<a>```?

Reply:
I think this test case - ```<var>bar</var>.<a>[[f oo]]</a>``` will fail the current logic. Maybe we could change the return condition to ```return !isInternalSlot || isPrevVar;```
So the filter function return true if the element is not an internal slot or if the previous element is var.
What do you think?

Reply:
Could we do it with the RegExp somehow?

Reply:
to be clear, `<a>[[f oo]]</a>` shouldn't be classed as being an internal slot. So,  `isInternalSlot` should be false in the case above.

Reply:
ok cool. Can you check why the test is failing? I probably broke it :)